### PR TITLE
Javascript typing tutor

### PR DESCRIPTION
**Pull request by josephswpark**
https://user-images.githubusercontent.com/113474457/198680774-f0efc930-cc9f-4a7e-b251-3aa407311ff8.mp4

### CODE REVIEW

File: javascript-typing-tutor/main.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+var charNodes = document.querySelectorAll('span');
+var currentIndex = 0;
+document.addEventListener('keydown', typingTest);
+
+function typingTest(event) {
+  if (currentIndex === charNodes.length) {
+    alert('DONE! Restarting Test....');
+    window.location.reload();
```
Comment (codingcodymiller) [inline javascript-typing-tutor/main.js]: You should **_never_** use `location.reload()`, it reloads the page entirely, which is one of the most inefficient actions you can take. Instead, you should reset each variable value and reset the classes on all elements to bring the page back to how it was when the page first loaded

File: javascript-typing-tutor/main.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  if (event.keyCode !== 8) {
+    checkCharacterIsValid(event.key, currentIndex);
+  }
+  if (event.key === charNodes[currentIndex].innerHTML || event.keyCode === 8) {
```
Comment (codingcodymiller) [inline javascript-typing-tutor/main.js:9]: `event.keyCode === 8` 

What is this for?

File: javascript-typing-tutor/main.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  if (keyCode === 8 && currentIndex > 0) {
+    currentIndex -= 1;
+    resetCharacter(currentIndex);
+  } else {
+    currentIndex += 1;
+  }
+}
+function checkCharacterIsValid(currentChar, charIndex) {
+  var currentNode = charNodes[charIndex];
+  if (currentChar === currentNode.innerHTML) {
```
Comment (codingcodymiller) [inline javascript-typing-tutor/main.js]: You should use `textContent` rather than `innerHTML`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: javascript-typing-tutor/main.js (hunk 2 above)
Comment (josephswpark) [reply javascript-typing-tutor/main.js:9]: I used this for backspace key.  Am i not allowed to use that?

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**